### PR TITLE
feat: add `nodejs:16` supported runtime

### PR DESCRIPTION
**Pull request by shazron**
This is enabled in Adobe I/O Runtime but is not the default yet.

## How Has This Been Tested?

npm test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ const globby = require('globby')
 const path = require('path')
 const archiver = require('archiver')
 // this is a static list that comes from here: https://developer.adobe.com/runtime/docs/guides/reference/runtimes/
-const SupportedRuntimes = ['nodejs:10', 'nodejs:12', 'nodejs:14']
+const SupportedRuntimes = ['nodejs:10', 'nodejs:12', 'nodejs:14', 'nodejs:16']
```
Comment (tmathern) [inline src/utils.js:24]: Do you have a strategy when to stop supporting node versions?

Comment (shazron) [reply src/utils.js:24]: The strategy is that we follow what the Runtime server supports. Actually, we could improve it here by querying the server metadata (and cache the response after the first time) so this list is dynamic...

Comment (shazron) [reply src/utils.js:24]: related: https://github.com/adobe/aio-lib-runtime/pull/101

Comment (purplecabbage) [reply src/utils.js:24]: #101 is merged, it is a good utility function.
I would prefer that we do not check the server for supported runtimes on every action deploy though.  Can we do it in response to an error as a way to provide more details to the user?

Comment (shazron) [reply src/utils.js:24]: Good idea - we can output that only on error. 

That is one case - if they are running node-10 and the server doesn't support it anymore, we output the supported runtimes.

The other case - if our action is node-18, and the server supports it, but the CLI code hasn't been updated. Do we want to support a way to "override" this check? I would think we would quickly update the CLI code so this wouldn't really be an issue but perhaps node-18 is in a testing phase or something and we don't want to make it generally available. Maybe an env var?

Comment (shazron) [reply src/utils.js:24]: Note for the second case, if the server doesn't support node-10 and they override, the server will not accept it anyway on create/update via the API)

Comment (shazron) [reply src/utils.js:24]: I'll skip the override for now and just implement calling `getSupportedRuntimes` on error.

Comment (shazron) [reply src/utils.js:24]: done, see latest commit

**Review by moritzraho — APPROVED** [review]
LGTM

Out of scope right now, but for open source kindness I wonder if we could cache a map of `apihost` => `[runtimes]` instead of just a static list.

**Review by purplecabbage — APPROVED** [review]
I think this is good.  Much better now that we only call on error.

**Comment by shazron** [comment]
@moritzraho I added https://github.com/adobe/aio-lib-runtime/issues/102 for the enhancement

**Pull request merged**